### ISSUE TITLE
Caching / Ability to run tests without hitting Github

### ISSUE THREAD

**Issue by Macil** [flow-typed/flow-typed#542]
While working on some definitions I've been running `node cli/dist/cli.js run-tests testname` very frequently. I just hit this error:

```
Running definition tests in /Users/chris/Coding/flow-typed/definitions/npm...

Fetching all Flow binaries...
UNCAUGHT ERROR: {
  "code": 403,
  "message": "{\"message\":\"API rate limit exceeded for xxx.xxx.xxx.xxx. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)\",\"documentation_url\":\"https://developer.github.com/v3/#rate-limiting\"}"
}
```

For now I've just edited cli/src/commands/runTests.js's getOrderedFlowBinVersions to contain `return ['v0.37.0'];`. Maybe flow-typed should cache calls to github, have a flag to not hit github, or fall back to operating from the cache if it fails to hit github.

**Comment by DamonOehlman** [comment]
From having a look at the [current implementation](https://github.com/flowtype/flow-typed/blob/master/cli/src/commands/runTests.js#L75-L198) the list of flow releases is tied pretty tightly to the github API call.  I think if that single function was divided into two parts the logic that is desired (by me also) would be simpler to implement:

1. Attempt to fetch the list of recent releases from github, tolerant of failure (but messaged appropriately).

2.  Read the list of releases stored in the `.flow-bins-cache` (and sort) as the list of releases that should be tested against and execute tests against those versions.

If this is something that folks think is the appropriate way to deal with this, then I'd be happy to raise a PR for splitting that function up into the component parts.  A later PR could be created for optionally removing the github release check altogether.

**Comment by DamonOehlman** [comment]
@AgentME did you want to take a look at current master and see if the solution I implemented in #700 meets your needs.  It certainly took care of mine in an offline case - hooray for testing definitions when subjected to poor Internet coverage (i.e train journeys).

If so, I reckon we close this issue :)

**Comment by AgentME** [comment]
The automatic fallback is nice, but it might also be nice to be able to explicitly opt out of the github connection so I don't have to wait on the connection or get my IP rate-limited from github. It could be triggered by a -o / --offline parameter, or by an environment variable like `FLOW_TYPED_OFFLINE=true`.

**Comment by DamonOehlman** [comment]
I think this should be easy enough to implement also.  Admittedly when working on the train it was pretty much an immediate fallback but you are right, when it's an API rate limit issue or when you don't want to consume API calls while working that nice to have also.

Let me see how I go submitting another PR to address that also.
